source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,168 @@
+(nixos-vms)=
+
+# NixOS virtual machines
+
+One of the most important features of NixOS is the ability to configure the entire system declaratively, including packages to be installed, services to be run, as well as other settings and options.
+
+NixOS configurations can be used to test and use NixOS using a virtual machine, independent of an installation on a "bare metal" computer.
+
+:::{.important}
+A NixOS configuration is a Nix language function following the [NixOS module](https://nixos.org/manual/nixos/stable/index.html#sec-writing-modules) convention. 
+:::
+
+## What will you learn?
+
+This tutorial serves as an introduction creating NixOS virtual machines.
+Virtual machines are a practical tool for debugging NixOS configurations.
+
+## What do you need?
+
+- A working [Nix installation](install-nix) or [NixOS](https://nixos.org/manual/nixos/stable/index.html#sec-installation)
```

Comment:
This does not work on macOS because evaluating the configuration already fails (even if the QEMU stuff would work, which I doubt):
```
error: assertion '(stdenv).isLinux' failed

       at /nix/store/gfhjcl9ck77gk7zw21sk13pw7zw4307g-78342239aac6382af25986219ffb163ba79b155a/pkgs/os-specific/linux/kernel/generic.nix:71:1:

           70|
           71| assert stdenv.isLinux;
             | ^
           72|
(use '--show-trace' to show detailed location information)
```

```suggestion
- A working [Nix installation](install-nix) on Linux, or [NixOS](https://nixos.org/manual/nixos/stable/index.html#sec-installation)
```

Reply:
Thanks for testing on an obscure OS :-P 
lets see how this develops https://github.com/NixOS/nixpkgs/issues/108984